deeper/plan.md

```diff
@@ -0,0 +1,182 @@
+A({n a b})   = a
+A(n:@)       = I(1, (3 5 3), n)
+```
+
+### Execution (X)
+
+```txt
+X((f x), e)         = X(f,e)
+X(<p>, e)           = X(p,e)
+X({n a b}, e)       = R(a,e,b)
+X(0, (_ n a b))     = {N(n) N(a) F(b)}
+X(1, (_ p l a n x)) = P(p,l,a,n,E(x))
+X(2, (_ z p x))     = C(z,p,N(x))
+X(3, (_ x))         = N(x)+1
+X(4, (_ x))         = <F(x)>
+```
+
+These reduction rules, combined with the opcodes, define the operational semantics of PLAN. They determine how PLAN expressions are evaluated and how computation proceeds.
+
+## The EDSL Within PLAN
+
+PLAN contains an embedded domain-specific language (EDSL) for defining law bodies. This EDSL is not a separate language per se, but rather a specific way of using PLAN constructs to represent law bodies.
+
+The EDSL for law bodies has the following constructs:
+
+```txt
+Law = Self                -- self-reference
+    | Var Nat             -- reference bound name (using indices instead of names)
+    | Const PLAN          -- produce constant value
+    | (Law Law)           -- function application
+    | let Law in Law      -- let binding (adding a value to the end of the environment. may self-reference)
+```
+
+These constructs are encoded within PLAN using partially applied forms of PLAN's native opcodes:
+
+|---------|--------------------|
+| pattern | meaning            |
+|---------|--------------------|
+| (0 f x) | (f x)              |
+| (1 v b) | let v in b         |
+| (2 k)   | Const k            |
+| 0       | Self               |
+| 1       | var 1              |
+| 2       | var 2              |
+| x       | Const x (fallback) |
+|---------|--------------------|
+
+It's important to note that none of these patterns reduce under normal PLAN evaluation rules. This allows any valid AST of the EDSL to be stored using this encoding.
+
+The EDSL and PLAN proper are tightly intertwined:
+
+1. Within a law body, you can always drop back into regular PLAN either by using the `Const` construct or automatically when a part of the law body doesn't have meaning in the EDSL.
+2. Conversely, PLAN switches into the EDSL whenever a law has been applied to enough arguments.
+
+This dual-language structure allows PLAN to maintain a simple core while providing more expressive power for defining functions.
+
+## Reflexivity in PLAN
+
+In PLAN, any program can deconstruct and analyze any terms it encounters during execution. This means that PLAN programs have the ability to:
+
+1. Inspect the structure of data and code
+2. Make decisions based on the form of expressions
+3. Modify or generate new code dynamically
+
+This level of reflexivity is not common in many programming languages or evaluation models.
+
+## PLAN as a Comprehensive Data Structure
+
+While it's not entirely accurate to call PLAN a format for human-readable executables or binaries, this perspective highlights PLAN's unique position as a comprehensive data structure that bridges multiple domains of computation and storage.
```

Comment:
I think it's accurate to call it a format for human-readable executables (but not binaries). I guess Claude wrote this. Did it elaborate?

Reply:
Claude wrote this and I didn't catch it. fixing.

Reply:
resolved in https://github.com/deathtothecorporation/plunder-docs/pull/21/commits/df86146a0fefcb51eed38b523d477301b1c09e75